[BE] Retrieve user information

nspc97 opened the pull request:
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

Commit:
[BE] Retrieve user information
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

Added function for retrieving user information

Commit:
[BE] Retrieve user information
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

Added function for retrieving user information

Commit:
[BE] Retrieve user information
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

Added RegisterController

Commit:
[BE] Retrieve user information
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

Changed @Route("/api", name="user") to @Route("/api", name="register") in RegisterController

SpartakusMd commented on src/Repository/ActivityRepository.php:
```diff
@@ -22,7 +22,7 @@ public function __construct(RegistryInterface $registry)
 
     /**
-     * @Rest\Delete("/activities/delete/{id}")
+     * @Rest\Delete("/activities/{id}/delete")
```
`@Rest` annotations should be present only in controlers. Please remove.

SpartakusMd commented on src/Repository/UserRepository.php:
```diff
@@ -18,4 +20,19 @@ public function __construct(RegistryInterface $registry)
+
+    /**
+     * Create an User.
+     * @Rest\Post("/register")
```
Please remove `@Rest` annotation.

SpartakusMd commented on src/Repository/UserRepository.php:
```diff
@@ -18,4 +20,19 @@ public function __construct(RegistryInterface $registry)
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function register(User $entityToPersist): void
```
Rename to `save()` as you are saving an entity.

Commit:
[BE] Retrieve user information
https://trello.com/c/n8EpNFPf/8-be-retrieve-user-information

removed rest annotations from repositories
